assets/html/js/search.js

```diff
@@ -1,154 +1,198 @@
-  "my",
-  "neither",
-  "no",
-  "nor",
-  "not",
-  "of",
-  "off",
-  "often",
-  "on",
-  "or",
-  "other",
-  "our",
-  "own",
-  "rather",
-  "said",
-  "say",
-  "says",
-  "she",
-  "should",
-  "since",
-  "so",
-  "some",
-  "than",
-  "that",
-  "the",
-  "their",
-  "them",
-  "then",
-  "there",
-  "these",
-  "they",
-  "this",
-  "tis",
-  "to",
-  "too",
-  "twas",
-  "us",
-  "wants",
-  "was",
-  "we",
-  "were",
-  "what",
-  "when",
-  "who",
-  "whom",
-  "why",
-  "will",
-  "would",
-  "yet",
-  "you",
-  "your",
-]);
-
-let index = new minisearch({
-  fields: ["title", "text"], // fields to index for full-text search
-  storeFields: ["location", "title", "text", "category", "page"], // fields to return with search results
-  processTerm: (term) => {
-    let word = stopWords.has(term) ? null : term;
-    if (word) {
-      // custom trimmer that doesn't strip @ and !, which are used in julia macro and function names
-      word = word
-        .replace(/^[^a-zA-Z0-9@!]+/, "")
-        .replace(/[^a-zA-Z0-9@!]+$/, "");
-
-      word = word.toLowerCase();
-    }
+SEARCH WORKER:
+
+Import minisearch
+
+Build index
+
+On message from main thread
+  run search
+  find the first 200 unique results from each category, and compute their divs for display
+    note that this is necessary and sufficient information for the main thread to find the
+    first 200 unique results from any given filter set
+  post results to main thread
+
+MAIN:
+
+Launch worker
+
+Declare nonconstant globals (worker_is_running,  last_search_text, unfiltered_results)
+
+On text update
+  if worker is not running, launch_search()
+
+launch_search
+  set worker_is_running to true, set last_search_text to the search text
+  post the search query to worker
+
+on message from worker
+  if last_search_text is not the same as the text in the search field,
+    the latest search result is not reflective of the latest search query, so update again
+    launch_search()
+  otherwise
+    set worker_is_running to false
+
+  regardless, display the new search results to the user
+  save the unfiltered_results as a global
+  update_search()
 
-    return word ?? null;
-  },
-  // add . as a separator, because otherwise "title": "Documenter.Anchors.add!", would not find anything if searching for "add!", only for the entire qualification
-  tokenize: (string) => string.split(/[\s\-\.]+/),
-  // options which will be applied during the search
-  searchOptions: {
-    prefix: true,
-    boost: { title: 100 },
-    fuzzy: 2,
+on filter click
+  adjust the filter selection
+  update_search()
+
+update_search
+  apply search filters by looping through the unfiltered_results and finding the first 200
+    unique results that match the filters
+
+  Update the DOM
+*/
+
+/////// SEARCH WORKER ///////
+
+function worker_function(documenterSearchIndex, documenterBaseURL, filters) {
+  importScripts(
+    "https://cdn.jsdelivr.net/npm/minisearch@6.1.0/dist/umd/index.min.js"
+  );
```

Comment:
Just a note: kinda mixed feelings about hard-coding dependencies in the JS in general (we declare them in the Julia code normally). It's fine here, since (1) I'm not sure there's an obvious way to reconcile this with the RequireJS thing we do right now, and (2) this is one dependency used in one place and not re-used in other JS files.

As a complete tangent: I do think we want to reconsider the RequireJS thing in general. It feels like it probably has outlived its usefulness. Feels like we should be able to just native modules or something.

Reply:
I also don't love this line.